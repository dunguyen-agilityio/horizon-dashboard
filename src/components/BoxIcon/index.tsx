import { cn } from '@nextui-org/theme';

interface IBoxIconProps {
  icon: JSX.Element;
  customClass?: string;
  onClick?: () => void;
}

const BoxIcon = ({
  icon,
  customClass = 'fill-secondary dark:fill-white',
  onClick,
}: IBoxIconProps) => (
  <i
    className={cn(customClass, 'w-6 h-6 flex justify-center items-center')}
    onClick={onClick}
  >
    {icon}
  </i>
);

export default BoxIcon;
