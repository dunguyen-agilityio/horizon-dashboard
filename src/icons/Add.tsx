interface AddProps {
  color?: 'primary' | 'secondary';
}

const classes = {
  primary: '[&>path]:stroke-primary [&>path]:dark:stroke-white',
  secondary: '[&>path]:stroke-white [&>path]:dark:stroke-primary',
};

const Add = ({ color = 'primary' }: AddProps) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="19"
    height="17"
    viewBox="0 0 19 17"
    fill="none"
    className={classes[color]}
  >
    <path
      d="M9.32703 3.71619V13.2787"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M14.3896 8.49744H4.26459"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);

export default Add;
