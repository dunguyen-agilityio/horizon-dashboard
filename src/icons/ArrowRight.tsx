const ArrowRight = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width={7} height={12}>
    <path d="M6.71 2.12 2.83 6l3.88 3.88a.996.996 0 1 1-1.41 1.41L.71 6.7a.996.996 0 0 1 0-1.41L5.3.7a.996.996 0 0 1 1.41 0c.38.39.39 1.03 0 1.42Z" />
  </svg>
);
export default ArrowRight;
