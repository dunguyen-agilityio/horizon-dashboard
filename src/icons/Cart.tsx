const Cart = () => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width={20}
    height={20}
    data-testid="cart-icon"
  >
    <path d="M14.55 11c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0 0 19.01 2H4.21l-.94-2H0v2h2l3.6 7.59-1.35 2.44C3.52 13.37 4.48 15 6 15h12v-2H6l1.1-2h7.45ZM5.16 4h12.15l-2.76 5H7.53L5.16 4ZM6 16c-1.1 0-1.99.9-1.99 2S4.9 20 6 20s2-.9 2-2-.9-2-2-2Zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2Z" />
  </svg>
);
export default Cart;
