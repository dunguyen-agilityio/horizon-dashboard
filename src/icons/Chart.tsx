const SvgComponent = () => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width={18}
    height={18}
    fill="none"
    className="[&>path]:fill-secondary [&>path]:dark:fill-white"
  >
    <path d="M2.467 5.733H2.7c.898 0 1.633.735 1.633 1.634v8.166c0 .899-.735 1.634-1.633 1.634h-.233a1.638 1.638 0 0 1-1.634-1.634V7.367c0-.899.735-1.634 1.634-1.634ZM9 .833c.898 0 1.633.735 1.633 1.634v13.066c0 .899-.735 1.634-1.633 1.634a1.638 1.638 0 0 1-1.633-1.634V2.467c0-.899.735-1.634 1.633-1.634Zm6.533 9.334c.899 0 1.634.735 1.634 1.633v3.733c0 .899-.735 1.634-1.634 1.634a1.638 1.638 0 0 1-1.633-1.634V11.8c0-.898.735-1.633 1.633-1.633Z" />
  </svg>
);
export default SvgComponent;
