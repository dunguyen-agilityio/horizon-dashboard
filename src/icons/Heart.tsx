const Heart = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width={16} height={15}>
    <path d="M13.944 1.4C11.959-.008 9.509.65 8.186 2.261 6.863.65 4.413-.016 2.43 1.401 1.376 2.15.715 3.42.669 4.757.566 7.795 3.15 10.23 7.097 13.963l.075.07c.572.54 1.451.54 2.022-.007l.083-.079c3.946-3.725 6.524-6.16 6.427-9.197-.046-1.33-.707-2.598-1.76-3.35ZM8.26 12.797l-.075.078-.075-.078C4.533 9.423 2.173 7.192 2.173 4.93c0-1.565 1.128-2.74 2.63-2.74 1.158 0 2.286.776 2.684 1.848h1.406c.39-1.072 1.518-1.847 2.675-1.847 1.504 0 2.631 1.174 2.631 2.74 0 2.261-2.36 4.492-5.938 7.866Z" />
  </svg>
);
export default Heart;
