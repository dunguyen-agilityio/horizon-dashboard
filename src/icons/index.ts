export { default as Add } from './Add';
export { default as ArrowRight } from './ArrowRight';
export { default as Cart } from './Cart';
export { default as Chart } from './Chart';
export { default as Cloud } from './Cloud';
export { default as EyeFilled } from './EyeFilled';
export { default as EyeSlashFilled } from './EyeSlashFilled';
export { default as Home } from './Home';
export { default as Kanban } from './Kanban';
export { default as Lock } from './Lock';
export { default as Menu } from './Menu';
export { default as More } from './More';
export { default as Moon } from './Moon';
export { default as Note } from './Note';
export { default as Notify } from './Notify';
export { default as Pen } from './Pen';
export { default as Search } from './Search';
export { default as UploadFile } from './UploadFile';
export { default as User } from './User';
