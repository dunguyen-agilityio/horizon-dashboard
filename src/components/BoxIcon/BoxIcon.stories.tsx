import { fn } from '@storybook/test';
import type { Meta, StoryObj } from '@storybook/react';

// Components
import BoxIcon from '.';

// Icons
import { Kanban } from '@/icons';

const meta = {
  title: 'Components/BoxIcon',
  component: BoxIcon,
  tags: ['autodocs'],
} satisfies Meta<typeof BoxIcon>;

export default meta;
type Story = StoryObj<typeof meta>;

export const BoxIconWithIcon: Story = {
  args: {
    icon: <Kanban />,
  },
};

export const BoxIconWithCustomClass: Story = {
  args: {
    ...BoxIconWithIcon.args,
    customClass: 'dark:fill-red-800, fill-green-800',
  },
};

export const BoxIconWithOnClick: Story = {
  args: {
    ...BoxIconWithCustomClass.args,
    onClick: fn(),
  },
};
