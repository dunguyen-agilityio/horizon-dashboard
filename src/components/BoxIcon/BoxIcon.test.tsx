import { fireEvent, render } from '@testing-library/react';

// Components
import BoxIcon from '.';

// Icons
import { Kanban } from '@/icons';

const onClickBoxIcon = jest.fn();

describe('BoxIcon component', () => {
  it('Should render BoxIcon match snapshot', () => {
    const { container } = render(<BoxIcon icon={<Kanban />} />);
    expect(container).toMatchSnapshot();
  });

  it('Should trigger onClick when simulator click in BoxIcon', () => {
    const { getByTestId } = render(
      <BoxIcon icon={<Kanban />} onClick={onClickBoxIcon} />,
    );

    const BoxIconName = getByTestId('kanban-icon');
    expect(BoxIconName).toBeInTheDocument();

    fireEvent.click(BoxIconName);
    expect(onClickBoxIcon).toHaveBeenCalled();
  });
});
