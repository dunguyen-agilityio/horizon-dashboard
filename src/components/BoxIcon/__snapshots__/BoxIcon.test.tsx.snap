// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxIcon component Should render BoxIcon match snapshot 1`] = `
<div>
  <i
    class="fill-secondary dark:fill-white w-6 h-6 flex justify-center items-center"
  >
    <svg
      data-testid="kanban-icon"
      height="18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 10h6c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1H1C.45 0 0 .45 0 1v8c0 .55.45 1 1 1Zm0 8h6c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1H1c-.55 0-1 .45-1 1v4c0 .55.45 1 1 1Zm10 0h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1ZM10 1v4c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1Z"
      />
    </svg>
  </i>
</div>
`;
